api/app/controllers/mno_enterprise/jpi/v1/admin/organizations_controller.rb

```diff
@@ -117,6 +122,78 @@ def invite_member
       @user = user.confirmed? ? invite : user
     end
 
+    def download_batch_example
+      path = File.join(File.dirname(File.expand_path(__FILE__)), '../../../../../assets/batch-example.csv')
+      send_file(path, filename: 'batch-example.csv', type: 'application/csv')
+    end
+
+    # POST /mnoe/jpi/v1/admin/organization/batch_import
+    # Organizations and Users batch import endpoint
+    # The file will be entirely rejected if one line has a validation failure
+    # Each line can be used to create a new customer organisation or a new customer user.
+    # Please use an external reference (external_id) to identify a company if you wish to add multiple users to the same company
+    # the import tool will not identify organisation based on names and will create a new organisation if no external reference is used.
+    # When an external reference if found, the organisation record is updated.
+    # If user email address already exists, it will be linked to the customer
+    def batch_import
```

Comment:
All this logic should be moved into another class (eg: `ImportCsvOrganization`)

`Controller`
```ruby
def batch_import
  # Check mandatory param(s)
  # Call CSV importer
  @results = ImportCsvOrganization.process(params[:file])
  render 'batch_import'
rescue =>e
  render_bad_request('batch_import', e.message)
end
```

`ImportCsvOrganization`
```ruby
def self.process(file_content)
  # validate and raise error
  # process lines
end
```

Reply:
Then you can easily unit test different scenarios

Reply:
Yes. I wish to still be able to return the errors as an array, as it's what is being displayed on the front-end.